### PR TITLE
Add tooltip to cover image button describing optimal image ratio

### PR DESCRIPTION
**Pull request by Zhao-Andy**
## What type of PR is this? (check all applicable)
- [x] Feature

## Description
This adds a tooltip to the cover image button in the "Rich + Markdown" editor when hovered over: "Use a ratio of 100:42 for best results."

https://user-images.githubusercontent.com/17884966/170611684-de34f301-7465-4d72-85a1-d9fe9ddb6d6f.mp4

It also adds the same tooltip as an inline comment under the cover image text for the "Basic" editor (v1 editor):
Screenshot

## Related Tickets & Documents
Closes https://github.com/forem/forem/issues/17769

## QA Instructions, Screenshots, Recordings
1. Open the editor
2. Hover over the Add Cover Image button
3. See that the tooltip appears

### UI accessibility concerns?
Yes, to be addressed as a separate comment below.

## Added/updated tests?
- [x] Yes

## [Forem core team only] How will this change be communicated?
- [x] I will share this change internally with the appropriate teams

## [optional] Are there any post deployment tasks we need to perform?
Nope

### CODE REVIEW

**Comment by Zhao-Andy** [comment]
There's currently an accessibility concern here. It currently doesn't work when `focused`. This is because the current implementation has a `<label>` that wraps an `<input>`, so when <kbd>TAB</kbd> key pressing through the elements, the `<input>` is focused on. The tooltip is attached as an HTML class to the `<label>`, so since it's not the element that's being focused, it won't show the tooltip.

This implementation is also missing some classes that I added manually to test the `focus` that is not in the PR right now because it was definitely a hack instead of a good solution.

So, my question for @aitchiss and @ludwiczakpawel: Are we okay with the tooltip not appearing when focused? Or should I go ahead and try to refactor this `<label>` and `<input>` combination into a Crayons `<Button>` component so that we can use the native Crayons tooltip option? Or maybe some other implementation? Would appreciate thoughts from either/both of you! 🙂

File: app/javascript/article-form/components/ArticleCoverImage.jsx [hunk 1]
```diff
@@ -46,6 +46,9 @@ const StandardImageUpload = ({
           className="screen-reader-only"
           data-max-file-size-mb="25"
         />
+        <span data-testid="tooltip" className="crayons-tooltip__content">
+          Use a ratio of 100:42 for best results.
+        </span>
```
Comment (Zhao-Andy) [inline app/javascript/article-form/components/ArticleCoverImage.jsx:51]: [nitpick, non-blocking]

Not in love with this copy. All suggestions appreciated.

File: app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx [hunk 2]
```diff
@@ -95,7 +95,7 @@ describe('<ArticleCoverImage />', () => {
       expect(queryByText(/uploading.../i)).not.toBeInTheDocument();
       expect(queryByLabelText('Add a cover image')).not.toBeInTheDocument();
       expect(getByRole('img', { name: 'Post cover' })).toBeInTheDocument();
-      expect(getByLabelText('Change')).toBeInTheDocument();
+      expect(getByLabelText('Change', { exact: false })).toBeInTheDocument();
```
Comment (Zhao-Andy) [inline app/javascript/article-form/components/__tests__/ArticleCoverImage.test.jsx:98]: Not sure if `{ exact: false }` is the best use here? Maybe using regex would be better? For example:
```js
getByLabelText(/Change.*/i)
```

**Review by jeremyf — APPROVED** [review]

**Comment by aitchiss** [comment]
> So, my question for @aitchiss and @ludwiczakpawel: Are we okay with the tooltip not appearing when focused? Or should I go ahead and try to refactor this <label> and <input> combination into a Crayons <Button> component so that we can use the native Crayons tooltip option? Or maybe some other implementation? Would appreciate thoughts from either/both of you! 🙂

@Zhao-Andy - just catching up on this as I've been OOO - I think we should create a new issue to make sure keyboard users can access this information too. Maybe it would make a good a11y bug for a community contribution? In terms of a solution, I wonder if we can do some special-case CSS magic that would allow the tooltip to show when the input has focus 🤔  It would be nice to keep the semantics of the `label` + `input` that we have now if we can.

As a small aside - did we consider adding the helper text as static text below the button (instead of in a tooltip)? Just thinking it's easier for people to be able to refer to visible description text as they're choosing their image rather than hold it in mind after the tooltip disappears. Just a thought though.

Let me know if I can help with creating that a11y issue!

**Pull request merged**